frontend/packages/topology/src/behavior/withRemoveConnector.tsx

```diff
@@ -38,16 +48,36 @@ export const withRemoveConnector = <P extends WithRemoveConnectorProps & Element
 ) => (WrappedComponent: React.ComponentType<P>) => {
   const Component: React.FC<Omit<P, keyof WithRemoveConnectorProps>> = (props) => {
     const [show, setShow] = React.useState(false);
+    const [startPoint, setStartPoint] = React.useState<Point | undefined>(undefined);
+    const [endPoint, setEndPoint] = React.useState<Point | undefined>(undefined);
     const onShowRemoveConnector = React.useCallback(() => setShow(true), []);
     const onHideRemoveConnector = React.useCallback(() => setShow(false), []);
 
+    const removeEdgeRef = useCallbackRef(
+      React.useCallback((node: SVGElement | null) => {
+        if (node instanceof SVGPathElement) {
+          const pathLength = node.getTotalLength();
+          const point = node.getPointAtLength(pathLength / 2);
+          setStartPoint(new Point(point.x, point.y));
+          setEndPoint(new Point(point.x, point.y));
+        }
+        if (node instanceof SVGLineElement) {
+          const start = new Point(node.x1.baseVal.value, node.y1.baseVal.value);
+          const end = new Point(node.x2.baseVal.value, node.y2.baseVal.value);
+          setStartPoint(start);
+          setEndPoint(end);
+        }
+      }, []),
+    );
+
     return (
       <WrappedComponent
         {...(props as any)}
         onShowRemoveConnector={onShowRemoveConnector}
         onHideRemoveConnector={onHideRemoveConnector}
+        removeEdgeRef={removeEdgeRef}
       >
-        {show && renderRemove(props.element, onRemove)}
+        {show && renderRemove(props.element, onRemove, startPoint, endPoint)}
```

Comment:
Maybe default to this behavior instead:
```suggestion
        {show && renderRemove(props.element, onRemove, startPoint || props.element.getStartPoint(), endPoint || props.element.getEndPoint())}
```